src/sknnr/transformers/_base.py

```diff
@@ -11,3 +11,25 @@ def fit(self, X, y=None, sample_weight=None):
         scaler = super().fit(X, y, sample_weight)
         scaler.scale_ = np.std(np.asarray(X), axis=0, ddof=self.ddof)
         return scaler
+
+
+class ComponentReducerMixin:
+    """
+    Mixin for transformers that allow reduction of the number of components.
+    """
+
+    def __init__(self, n_components=None):
+        self.n_components = n_components
+
+    def set_components(self, ordination_obj):
```

Comment:
I haven't yet done any typing on these functions.  `ordination_obj` here is an instance of `CCA` or `CCorA`, so probably just needs a union type?

Reply:
Absolutely, this is the right call.  I went with `ordination_` just to be a bit more explicit.